### PR TITLE
Case classes and immutability

### PR DESCRIPTION
**Pull request by rxin**
Added a new section, since I've been seeing more mutable fields in case classes.

### CODE REVIEW

**Comment by rxin** [comment]
cc @marmbrus @aarondav

File: README.md [hunk 1]
```diff
@@ -412,6 +413,23 @@ list.map({ item => ... })
...
+- Copy constructor
+- Pattern matching on constructor parameters
+- Automatic toString/hash/equals implementation
+
+Constructor parameters should NOT be mutable for case classes.
+```scala
+// This is OK
+case class Person(name: String, age: Int)
+
+// This is NOT OK
```
Comment (aarondav) [inline README.md]: Could you give a reason behind this?

Comment (rxin) [inline README.md]: @marmbrus ?

Comment (marmbrus) [inline README.md]: I actually have do this.  I think its only a bad idea if the case class is not local (and in general I think sharing mutable data structures across boundaries is a bad idea)

Comment (aarondav) [inline README.md]: That does sound like a more general point, potentially exceeding a style guide, but we can at least give it as the reasoning. I thought there might also be a more Scala-y reason like you'll mess up all hash maps that contain this as a key :)

Comment (rxin) [inline README.md]: Yea the hash map thing is good. I can update it.

**Comment by rxin** [comment]
I've updated this. PTAL.

**Comment by rxin** [comment]
@HyukjinKwon and @Hawstein - want to update the Korean / Chinese translations?

**Comment by HyukjinKwon** [comment]
Sure, thanks for pinging.

**Comment by Hawstein** [comment]
@rxin Sure.

**Pull request merged**